lib/order.rb

```diff
@@ -0,0 +1,43 @@
+class Order
+  
+  attr_reader :total
+  attr_accessor :status
+  
+  def initialize
+    @status = Array.new
+    @total = 0
+  end
+  
+  def add(item)
+    status << item
+    calculate_partial
+  end
+  
+  def check_order
+    calculate_basket
+    calculate_total
+  end
+  
+  def clear_order
+    self.status = Array.new
+    "Your basket is now empty."
+  end
+  
+  private
+  
+  def calculate_basket
+    basket = status.group_by { |item| item.keys }
+    basket.each_pair {|item, number| puts "#{item.join} x #{number.size} = £#{(number[0].values.join.to_i) * (number.size/1.0)}"}
```

Comment:
Metrics/LineLength: Line is too long. [129/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/RedundantParentheses: Don't use parentheses around a method call.